holidays/groups/custom.py

```diff
@@ -18,7 +22,7 @@ class StaticHolidays:
     an external class.
     """
 
-    def __init__(self, cls) -> None:
+    def __init__(self, cls, inter_year: bool = False) -> None:
```

Comment:
I don't think the overhead worth introducing the flag, we can check it for all entities.